### ISSUE TITLE
Does not seems to autocomplete scope packages in package.json

### ISSUE THREAD

**Issue by jpolo** [bali182/autocomplete-json#65]
On latest release, it seems that autocompleting is broken for this kind of file : 
```json
{
  "devDependencies": {
     "@myscope/mypackage": "<cursor position>",
  }
}
```
It works for regular packages so it might be just a problem with eight @ or / character

**Comment by bali182** [comment]
I don't think it was ever supported, not intentionally at least. Will look into it.

**Comment by bali182** [comment]
Do scoped packages mean `@<some-vendor>/<per-vendor-package-name>`? If so the first step would be to figure out  if npm has any APIs from which the vendors and per-vendor packages are queryable - otherwise it's impossible. Now it was quite a lot of research to figure out how to query their rest API, @jpolo if you could provide any help with this new bit of  info I'd be really grateful.

**Comment by jpolo** [comment]
Yes it is `@<some-vendor>/<per-vendor-package-name>`.
After digging, here is what I found : 
* package name must be prefixed with @ in the package.json. 
* package name must NOT have @ in some commands (not in every, it's only because of a conflict with `package@version` scheme) 

Example : 
`npm show "@rdbird/eslint-config" versions` => FAIL
`npm show "rdbird/eslint-config" versions` => OK

I hope this would help, if not I will dig deeper...

**Comment by bali182** [comment]
Great, thank you! One more thing I'm not sure about is the following. You are typing and you are here:

```"@ca|"```

(`|` is the cursor). What would you expect? Autocomplete for all the vendors (list of `car`, `cat`, `call` etc - let's say these are vendors) or autocomplete for every package that starts with `@ca` (list of `@cat/foo`, `@cat/bar`, `@car/x`, `@call/y`)?

Similarly for the package part

```"@cat/f|"``` 

Would you expect for the search to semantically be aware that you are looking for packages that belong to the `cat` vendor and starts with `f`, or would you expect to bluntly search through all the packages which starts with `@cat/f`?

**Comment by jpolo** [comment]
Very often vendor has few packages and there might be lots of vendors. I
would rather expect that autocompletion is a list of ordered fullnames
rather than a two steps vendors then package list.

**Comment by bali182** [comment]
Ok let's start with that, later we can refine it :+1:

**Comment by bali182** [comment]
Looked into what's the problem: It's the npm registry.
The URL I'm currently using to get the packages is this ([here is the source](https://github.com/bali182/npm-package-lookup/blob/master/index.js#L11)):

https://skimdb.npmjs.com/registry/_design/app/_view/browseAll?group_level=1&limit=50&start_key=["<prefix>"]&end_key=["<prefix>z",{}]

So for example if you type `babel`, the URL looks like this:
https://skimdb.npmjs.com/registry/_design/app/_view/browseAll?group_level=1&limit=50&start_key=["babel"]&end_key=["babelz",{}] - you can click it and see the results.

Now if you search for anything that starts with `@` the registry returns an empty result set:
https://skimdb.npmjs.com/registry/_design/app/_view/browseAll?group_level=1&limit=50&start_key=["@"]&end_key=["@z",{}].

We need to figure out how to query scoped packages. If you know anything about this plz HALP!

**Comment by bali182** [comment]
VScode seem to have the same issue, I'll open a ticket  there as well, see if they know how to query the scoped packages 

https://github.com/Microsoft/vscode/blob/cc8df7305bebe1559ac0a2af270bdf2daf518185/extensions/javascript/src/features/packageJSONContribution.ts#L54

**Comment by UziTech** [comment]
It looks like vscode hard codes a list of scope names for scoped packages.

If we want this feature we will have to do something similar.

https://github.com/microsoft/vscode/blob/8d29dde51cf515df0061f086f0b9c8e3c2513526/extensions/npm/src/features/packageJSONContribution.ts#L29

**Comment by UziTech** [comment]
I think this is fixed in #114 by using `api.npms.io` instead of `skimdb.npmjs.com` to search for packages.

**Comment by UziTech** [comment]
Should be fixed in v5.5.5
